backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala

```diff
@@ -110,6 +111,60 @@ object BackendSettings extends BackendSettingsApi {
     }
   }
 
+  override def supportFileFormatWrite(format: FileFormat, fields: Array[StructField]): Boolean = {
+    // Validate if all types are supported.
+    def validateTypes(fields: Array[StructField]): Boolean = {
+
+      val unsupportedDataTypes =
+        fields.flatMap {
+          field =>
+            field.dataType match {
+              case _: TimestampType => Some("TimestampType")
+              case struct: StructType if validateTypes(struct.fields) =>
```

Comment:
Should't this be **if !validateTypes(struct.fields)**

Reply:
@ayushi-agarwal The return value is the unsupported type.